### PR TITLE
Fixup: changed default cpu model to 'host' as it pickup unknown model

### PR DESCRIPTION
**Pull request by sathnaga**
Fixup: changed default cpu model to 'host' as it pickup unknown model

### CODE REVIEW

**Comment by sathnaga** [comment]
@yangdongsheng @ypu @cevich Please review and push it, Thanks in advance. :-)

File: shared/cfg/machines.cfg [hunk 1]
```diff
@@ -15,6 +15,8 @@ variants:
     - @pseries:
         only ppc64
         machine_type = pseries
+        auto_cpu_model = "no"
+        cpu_model = host
```
Comment (yangdongsheng) [inline shared/cfg/machines.cfg]: As I am not familiar with the config of "auto_cpu_model", sorry I can not say more about it here. It seems this is designed special for qemu test. The only place it's used by:
```
    if (params.get("auto_cpu_model") == "yes" and
            vm_type == "qemu"):
```

But from the commit message, I think you got a unexpected value from ```utils_misc.get_qemu_best_cpu_model```, why not to change this function to return a correct value in default case? Just a question from curious.

**Comment by cevich** [comment]
<--- also not familiar :confused:

Maybe on the off-chance, @ldoktor knows something about this?  IIRC he is or was working on ppc64 stuff.

**Comment by ypu** [comment]
Hi seems currently the cpu_model for ppc is hard code in the function and is kind of out of date.So this patch should work. But suggest to use parameter "default_cpu_model" for it, and here is the doc for it:

    # If no host cpu model can be found on qemu_cpu_models, choose the default
    return params.get("default_cpu_model", "qemu64")

**Comment by ldoktor** [comment]
Hello @sathnaga, it's been ages since I dealt with `auto_cpu_model`. Yes, it's not ideal and even less useful for powerpc. Anyway using `host` seems even less useful, according to http://wiki.qemu.org/Features/CPUModels#-cpu_host_and_feature_probing `-cpu host` enables everything which is supported __OR__ can be emulated, and that is probably not what we want.

So from my point of view we can:

1. split `pseries` to `power7`, `power8`, `power8le`..., hardcode the cpu type and let users execute the correct variant.
2. implement cpu detection mechanism (at least simplistic)
3. say we don't care about emulated flags and use `host`

Anyway the feature owner is @ehabkost. Can I ask you, Eduardo, to give us some details about `-cpu` on `ppc64`? Any advises are welcome.

**Comment by ehabkost** [comment]
@ldoktor the wiki page is about x86 mostly. For now, `host` in x86 has a different meaning from `host` on ppc64.

I don't know what are the consequences of using `host` on each of the PPC machines in QEMU, so I can't tell if this is a good idea or not. But, considering that this discussion is just about choosing a  reasonable default (as we still need to be able to ask virt-test to test any CPU model), isn't it better to simply omit the `-cpu` option by default and let QEMU choose the default model for the machine?

**Comment by ldoktor** [comment]
Thanks @ehabkost, working on arm I forgot we can omit the `-cpu`. @sathnaga would have anything against `cpu_model =`? I just tested it and it works fine.

**Comment by ldoktor** [comment]
@sathnaga I thought about Eduardo comment and probably it's better to remove `-cpu` when our detection mechanism fails and there is no `params` default. What do you think about it?

**Comment by ehabkost** [comment]
@ldoktor could you clarify what you mean by "when our detection mechanism fails"? Which detection mechanism, and what exactly is considered a failure for it?

**Comment by ldoktor** [comment]
Well there is a detection mechanism used to pick suitable cpu. https://github.com/autotest/virt-test/blob/master/virttest/utils_misc.py#L1777 here is how we detect host-supported-cpus, then qemu-supported-cpus are listed and then the most powerful is chosen. Anyway for `ppc` this is simply skipped and 'POWER7' (if supported) is used. Otherwise default is used (qemu64).

**Comment by ldoktor** [comment]
Btw there is also `default_cpu_model`, so instead of disabling autodetection you can just set `default_cpu_model = host` in `pseries` section. That way even if better logic is developed for PPC we can get better results while keeping good default.

**Comment by sathnaga** [comment]
@ehabkost @ldoktor : like the idea of having the qemu to take care of choosing the cpu model.
Tested it is working fine too, till we have the finer way of virt-test choosing the best cpu model for the ppc, it would be better to have this way, anyways if user want, can also choose by config as well using cpu_model and auto_cpu_model params.

**Comment by ldoktor** [comment]
So do you agree with https://github.com/autotest/virt-test/pull/2148 ? IMO it does exactly what you stated here.

**Comment by sathnaga** [comment]
@ldoktor : Yes, agree :+1:

**Comment by sathnaga** [comment]
Addressed as part of https://github.com/autotest/virt-test/pull/2148 and hence closing this.

**Pull request closed without merging**